config/cmake/HDF5UseFortran.cmake

```diff
@@ -260,7 +198,7 @@ foreach (KIND ${VAR})
           USE ISO_C_BINDING
           IMPLICIT NONE
           INTEGER (KIND=${KIND}) a
-          OPEN(8,FILE='pac_validIntKinds.out',FORM='formatted')
+          OPEN(8,FILE='pac_validIntKinds.${KIND}.out',FORM='formatted')
```

Comment:
Each iteration generates its own output file. As check_fortran_source_runs only executes one time, n + 1 configure iterations would read the last output file of the first run, so the type sizes were always 16, tripping an assert in the H5match_types generator at compile time.